File: problems/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// Prototypes
void get_data(int data[], string text);

int main(void)
{
    string text = get_string("Text: ");
    // Letters, words, sentences 
    int data [] = {0, 1, 0};

    get_data(data, text);
    float L = (data[0] * 100.0) / data[1]; // Letters * 100 / words
    float S = (data[2] * 100.0) / data[1]; // Sentences * 100 / words
    float index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }
}

void get_data(int data[], string text)
{
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            data[0]++;
        }
        if (text[i] == ' ')
        {
            data[1]++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            data[2]++;
        }
    }
}

File: problems/sentimental/readability/readability.py
# Import get string
from cs50 import get_string

# Get text from user
text = get_string("Text: ")

# Letters, words, sentences
# words starts in 1 because there is a "missing" ' ' (space) after the last word
data = [0, 1, 0]

# Go trough the text counting letters words and sentences
for i in range(0, len(text)):
    if text[i].isalpha():
        data[0] += 1
    elif text[i].isspace():
        data[1] += 1
    elif (text[i] == '.' or text[i] == '!' or text[i] == '?'):
        data[2] += 1

# Evaluate L and S
L = (data[0] * 100.0) / data[1]  # Letters * 100 / words
S = (data[2] * 100.0) / data[1]  # Sentences * 100 / words

# Evaluate Coleman-Liau index
index = 0.0588 * L - 0.296 * S - 15.8
index = round(index, 0)

# Print the grade of the text
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(index))